### ISSUE TITLE
[Ivy] Subclass/component now inheriting `@Component({ host: {} })` properties from superclass.

### ISSUE THREAD

**Issue by nbeekman** [angular/angular#36563]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/core

### Is this a regression?

Yes, the previous version in which this bug was not present was: 8 or 9 with Ivy disabled

### Description

With Ivy enabled, subclasses extending a superclass now inherit properties from `@Component({})`. More specifically, subclasses now inherit `host` properties such as `(click)` and `(keydown)`. In previous versions and Angular 9 with Ivy disabled, this does not happen.

## 🔬 Minimal Reproduction

https://stackblitz.com/edit/angular9-class-inheritance

**Note:** Open the console and click on either Foo or Bar. Then, change the ‘enableIvy’ value in the ‘tsconfig.json’ file or in the StackBlitz settings menu on the left, rerun, and you'll see a difference in what’s outputted to the console.

## 🌍  Your Environment

**Angular Version:**
<pre><code>9.1.1</code></pre>

**Comment by Vizer** [comment]
Same issue here. It would be great having the possibility to override inherited bindings

P.S. my question on SO with no answer [link](https://stackoverflow.com/questions/60674461/angular-ivy-host-binding-inheritance)

**Comment by kara** [comment]
@nbeekman This is actually an intentional change (see the first entry under "less common changes" in the [Ivy compatibility guide](https://angular.io/guide/ivy-compatibility). 

Background: In View Engine (e.g. v8), `@HostBinding` would be inherited by subclasses, but equivalent bindings in `host` would not be, which could be confusing. In Ivy, we made it consistent so `@HostBinding` and `host` were actually equivalent.

That said, having a way to override host bindings sounds like an interesting use case, so I'll keep this issue open for discussion.

**Comment by creativeux** [comment]
Thank you for the feedback @kara - that was what we needed to get unblocked. Using your note combined with the conversation in [this open issue](https://github.com/angular/angular/issues/33632) I ended up removing the `host`-based click bindings in both superclass and subclass and moved it to `@HostListener` in the superclass and the inheritance became consistent between Ivy and View Engine.

I found myself in a "fix Ivy, break View Engine, fix View Engine, break Ivy" feedback loop while I was chasing this which was an awkward developer experience caused by the inconsistency in how `host` and `@HostListener` are interpreted. This could be an error between seat and keyboard on my end misunderstanding the difference between `host: { '(click)': 'doSomething()' }` and @HostListener('click') doTheThing() { doSomething(); }` but in my mental model they are equivalent, so the behavioral difference in an inheritance scenario is still confusing to me.

Thanks again for your response, it was right on. Happy to continue the discussion if it is helpful.
